### PR TITLE
[Featured] implemented 9-ride-request-schedule-tile

### PR DESCRIPTION
**Pull request by Pradeep-O**

### CODE REVIEW

File: lib/widgets/request_ride_schedule_tile.dart [hunk 1]
```diff
@@ -0,0 +1,115 @@
+import 'package:flutter/material.dart';
```
Comment (maheshalyana) [inline lib/widgets/request_ride_schedule_tile.dart:1]: Choose a proper tile name

File: lib/screens/home_screen_tab_structure.dart [hunk 2]
```diff
@@ -68,14 +78,18 @@ class _HomeScreenState extends State<HomeScreen> {
               Container(
                 height: height * 0.43,
                 margin: const EdgeInsets.symmetric(vertical: 20),
-                child: const TabBarView(children: [
-                  ScheduleTile(),
-                  RequestTile(),
+                child: TabBarView(children: [
+                  TileTwoButtons(values: values),
```
Comment (maheshalyana) [inline lib/screens/home_screen_tab_structure.dart:82]: both should be list view, that means, there should be scrolling of the tiles vertically

**Review by maheshalyana — CHANGES_REQUESTED** [review]

File: lib/widgets/request_ride_schedule_tile.dart (hunk 1 above)
Comment (Pradeep-O) [reply lib/widgets/request_ride_schedule_tile.dart:1]: Changing to RideScheduleTile

Comment (maheshalyana) [reply lib/widgets/request_ride_schedule_tile.dart:1]: how about `RideDetailsTile`

File: lib/screens/home_screen_tab_structure.dart [hunk 3]
```diff
@@ -136,49 +154,3 @@ class _HomeScreenState extends State<HomeScreen> {
     );
   }
 }
-
```
Comment (maheshalyana) [inline lib/screens/home_screen_tab_structure.dart:139]: Don't remove these Classes, instead add the widget class in these classes and use these classes in the tabs

**Review by maheshalyana — CHANGES_REQUESTED** [review]

File: lib/widgets/request_ride_schedule_tile.dart (hunk 1 above)
Comment (Pradeep-O) [reply lib/widgets/request_ride_schedule_tile.dart:1]: I will go with it sir.

**Review by maheshalyana — APPROVED** [review]

**Review by Abhi-0987 — COMMENTED** [review]
image

Padding is needed at the top..

File: test/home_screen_tab_structure_test.dart [hunk 4]
```diff
@@ -10,27 +11,25 @@ void main() {
...
     expect(find.text('Offer a Ride'), findsOneWidget);
     expect(find.text('Get a Ride'), findsOneWidget);
 
     await tester.tap(find.byType(TabBar));
     await tester.pumpAndSettle();
 
-    expect(find.text('Schedule Tile'), findsNothing);
-    expect(find.text('Requests Tile'), findsOneWidget);
+    expect(find.byType(ScheduleTile), findsNothing);
+    expect(find.byType(RideDetailsTile), findsOneWidget);
```
Comment (maheshalyana) [inline test/home_screen_tab_structure_test.dart:31]: What is there is not Schedule, then it will fail, so remove this one

**Review by maheshalyana — CHANGES_REQUESTED** [review]

File: test/home_screen_tab_structure_test.dart (hunk 4 above)
Comment (Pradeep-O) [reply test/home_screen_tab_structure_test.dart:31]: ScheduleTile has findsNothing in this.

Comment (Pradeep-O) [reply test/home_screen_tab_structure_test.dart]: Removed them

File: test/home_screen_tab_structure_test.dart [hunk 5]
```diff
@@ -10,27 +11,23 @@ void main() {
         home: HomeScreen(),
       ),
     );
-
     expect(find.text('RIDE BHAIYA'), findsOneWidget);
 
     expect(find.byType(TabBar), findsOneWidget);
 
-    expect(find.text('Latest Schedules'), findsOneWidget);
```
Comment (maheshalyana) [inline test/home_screen_tab_structure_test.dart:18]: Why did you removed these test cases

File: test/home_screen_tab_structure_test.dart [hunk 6]
```diff
@@ -10,27 +11,23 @@ void main() {
...
 
     expect(find.text('Offer a Ride'), findsOneWidget);
     expect(find.text('Get a Ride'), findsOneWidget);
 
     await tester.tap(find.byType(TabBar));
     await tester.pumpAndSettle();
 
-    expect(find.text('Schedule Tile'), findsNothing);
-    expect(find.text('Requests Tile'), findsOneWidget);
+    expect(find.byType(RequestTile), findsOneWidget);
```
Comment (maheshalyana) [inline test/home_screen_tab_structure_test.dart:30]: Remove this test case we don't need them

File: test/home_screen_tab_structure_test.dart [hunk 7]
```diff
@@ -10,27 +11,23 @@ void main() {
...
-    expect(find.text('Latest Schedules'), findsOneWidget);
-    expect(find.text('Latest Requests'), findsOneWidget);
-
     expect(find.byType(TabBarView), findsOneWidget);
 
-    expect(find.text('Schedule Tile'), findsOneWidget);
-    expect(find.text('Requests Tile'), findsNothing);
+    expect(find.byType(ScheduleTile), findsOneWidget);
+    expect(find.byType(RideDetailsTile), findsOneWidget);
+    expect(find.byType(RequestTile), findsNothing);
```
Comment (maheshalyana) [inline test/home_screen_tab_structure_test.dart:22]: remove these test cases

**Review by maheshalyana — CHANGES_REQUESTED** [review]

File: test/home_screen_tab_structure_test.dart (hunk 5 above)
Comment (Pradeep-O) [reply test/home_screen_tab_structure_test.dart:18]: There will be no text output for the classes as seen in the Figma.
Before modification, they were there to represent the tile, but after modification we just need the tile and its contents, no need for the Texts like 'Latest Schedules' and 'Latest Requests.

File: test/home_screen_tab_structure_test.dart (hunk 7 above)
Comment (Pradeep-O) [reply test/home_screen_tab_structure_test.dart:22]: Ok sir

File: test/home_screen_tab_structure_test.dart (hunk 5 above)
Comment (maheshalyana) [reply test/home_screen_tab_structure_test.dart:18]: I think these are being rendered in tab bar right?

Comment (Pradeep-O) [reply test/home_screen_tab_structure_test.dart:18]: Yeah got it

**Review by maheshalyana — APPROVED** [review]

**Review by Abhi-0987 — APPROVED** [review]
image
image
image

Screen matches with the design and test case passed..moving it to acceptance

**Pull request merged**